src/lib/autoinstall/xml_validator.rb

```diff
@@ -56,8 +59,29 @@ def valid?
       errors.empty?
     end
 
+    # Writes the errors to a file in yaml format
+    def store_errors
+      File.write(errors_file_path, errors.to_yaml)
+    end
+
+    # Checks whether the current errors have been already stored in a file or
+    # not
+    #
+    # @return [Boolean] whether the errors have been already stored or not
+    def errors_stored?
+      File.exist?(errors_file_path)
+    end
+
   private
 
+    def id
+      Digest::MD5.hexdigest(errors.to_yaml)
+    end
+
+    def errors_file_path
```

Comment:
@jreidinger @imobachgs @lslezak 

Maybe we would prefer to only store last errors reported instead of all the different ones, what do you think? If we follow the current approach, would we prefer to store them under `xml_validations` folder instead?

Reply:
do we really need to store complete errors? Why not store only that id ( md5sum ) of errors? so we just detect known or not known.

Reply:
After discuss it with @imobachgs we have decided to move that logic to the XmlChecks class using a singleton object which load its state from a file when initialized.